contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -415,12 +410,11 @@ contract MixinExchangeCore is
         //     order.makerAssetAmount * takerAssetFilledAmount
         // NOTE: This assertion should never fail, it is here
         //       as an extra defence against potential bugs.
-        require(
-            safeMul(makerAssetFilledAmount, order.takerAssetAmount)
-            <= 
-            safeMul(order.makerAssetAmount, takerAssetFilledAmount),
-            "INVALID_FILL_PRICE"
-        );
+        if (safeMul(makerAssetFilledAmount, order.takerAssetAmount)
+                >
+                safeMul(order.makerAssetAmount, takerAssetFilledAmount)) {
+            rrevert(FillError(orderInfo.orderHash, FillErrorCodes.INVALID_FILL_PRICE));
```

Comment:
Technically we'd need `takerAssetFilledAmount` and `makerAssetFilledAmount` here since without those values, one couldn't figure out what went wrong. However, this is a cautionary assertion that should never hit, so maybe we can pass on this one.